Core.Collectors/Web/HttpExceptionSignature.cs

```diff
@@ -8,22 +8,37 @@ namespace Microsoft.CloudMine.Core.Collectors.Web
 {
     public class HttpExceptionSignature
     {
-        public static readonly HttpExceptionSignature RequestTimeoutException = new HttpExceptionSignature(exception => 
-        {
-            Type exceptionType = exception.GetType();
-            return exceptionType == typeof(TaskCanceledException) && exception.Message.Equals("The operation was canceled.");
-        });
-
         private readonly Func<Exception, bool> matcher;
+        private readonly Func<Task> handler;
 
-        public HttpExceptionSignature(Func<Exception, bool> matcher)
+        public HttpExceptionSignature(Func<Exception, bool> matcher, Func<Task> handler = null)
         {
             this.matcher = matcher;
+            this.handler = handler;
         }
 
         public bool Matches(Exception exception)
         {
             return matcher.Invoke(exception);
         }
+
+        public Task Handle()
+        {
+            if (null != this.handler)
+            {
+                return this.handler.Invoke();
```

Comment:
>this.handler.Invoke(); [](http://example.com/codeflow?start=23&length=22)

Could you check whether this.handler() works (instead of explicit invoke)?